app/src/main/java/com/obsidian/plugins/task_notifier/os/NotificationManager.kt

```diff
@@ -12,19 +16,20 @@ import com.obsidian.plugins.task_notifier.R
 class NotificationManager {
   companion object {
     var NOTIFICATION_CHANNEL_ID = "OBSIDIAN_TASK_NOTIFICATIONS_ID"
+    var NOTIFICATION_FOREGROUND_CHANNEL_ID = "OBSIDIAN_TASK_FREGROUND_ID"
 
     @JvmStatic
-    fun notify(context: Context, title: String, text: String, reqId: Int = 1): Notification {
-      this.ensureNotificationChannelExists(context)
-      val notification = this.createNotification(context, title, text)
+    fun notify(context: Context, title: String, text: String, reqId: Int = 1, channel: String = NOTIFICATION_CHANNEL_ID): Notification {
```

Comment:
I think the knowledge regarding which channel should be used is in the scope of the NotificationManager. Maybe using an enum and then translating it to an actual notification channel would be better. What do you think about this idea?

I would call it NotificationRequestSource which could be service, file change, reminder trigger etc..


Also if we are implementing this the ID of the caller entity should be included also as a parameter like a guid or something like that.

Reply:
I'm not sure i understand it.
You want to change the parameter to like   "domain" what is come from an enum? And inside the notification manager we convert it to the right channel?

Reply:
Yeah something like that. The caller of the NotificationManager.notify should not know about channeles etc. That knowledge should stay in the NotificaitonManager itself